langchain4j-agentic/src/main/java/dev/langchain4j/agentic/internal/AgentInvoker.java

```
@@ -18,11 +21,14 @@ public interface AgentInvoker extends AgentSpecification {
 
     String toCard();
 
-    Object[] toInvocationArguments(AgenticScope agenticScope) throws MissingArgumentException;
+    AgentInvocationArguments toInvocationArguments(AgenticScope agenticScope) throws MissingArgumentException;
 
-    default Object invoke(Object agent, Object... args) throws AgentInvocationException {
+    default Object invoke(DefaultAgenticScope agenticScope, Object agent, AgentInvocationArguments args) throws AgentInvocationException {
         try {
-            return method().invoke(agent, args);
+            onInvocation(new AgentRequest(agenticScope, name(), args.namedArgs()));
```

What would/should happen if an exception will be thrown from the user-provided callback?
- fail the whole flow
- ignore the exception
- ignore the exception and log it
- somehting else?

---

I would say that an exception thrown in user's code is something that should be raised up to the user again and then make the whole flow to fail, but to be honest I'm not strongly opinionated on this. Any different point of view?

---

Considering that this is an observability feature (secondary priority), I am not sure whether an exception (for example, because of an observability backend is temporarily unreachable) should stop the flow (primary priority). User should definitely be notified though

---

Agreed, done.